tests/ops/service/connectors/test_fides_connector.py

```diff
@@ -0,0 +1,101 @@
+import uuid
+from typing import Tuple
+
+import pytest
+
+from fides.api.ops.graph.traversal import TraversalNode
+from fides.api.ops.models.connectionconfig import ConnectionTestStatus
+from fides.api.ops.models.policy import Policy
+from fides.api.ops.models.privacy_request import PrivacyRequest, PrivacyRequestStatus
+from fides.api.ops.schemas.connection_configuration.connection_secrets_fides import (
+    DEFAULT_POLLING_INTERVAL,
+    DEFAULT_POLLING_RETRIES,
+)
+from fides.api.ops.service.connectors.fides.fides_client import FidesClient
+from fides.api.ops.service.connectors.fides_connector import FidesConnector
+from tests.ops.graph.graph_test_util import generate_node
+
+
+@pytest.mark.unit
+class TestFidesConnectorUnit:
+    """
+    Unit tests against functionality in the FidesConnector class
+    """
+
+    def test_connector_attributes_assigned_defaults(
+        self, test_fides_connector: FidesConnector
+    ):
+        assert test_fides_connector.polling_interval == DEFAULT_POLLING_INTERVAL
+        assert test_fides_connector.polling_retries == DEFAULT_POLLING_RETRIES
+
+    def test_connector_attributes_assigned(
+        self,
+        test_fides_connector_overriden_polling: FidesConnector,
+        fides_connector_polling_overrides: Tuple[int, int],
+    ):
+        assert (
+            test_fides_connector_overriden_polling.polling_retries
+            == fides_connector_polling_overrides[0]
+        )
+        assert (
+            test_fides_connector_overriden_polling.polling_interval
+            == fides_connector_polling_overrides[1]
+        )
+
+    def test_create_client(self, test_fides_connector: FidesConnector):
+        client: FidesClient = test_fides_connector.create_client()
+        assert client.token is not None
+        assert client.uri == test_fides_connector.configuration.secrets["uri"]
+        assert client.username == test_fides_connector.configuration.secrets["username"]
+        assert client.password == test_fides_connector.configuration.secrets["password"]
+
+
+@pytest.mark.integration
+class TestFidesConnectorIntegration:
+    """
+    Integration tests against functionality in the FidesConnector class
+    that interacts with a running Fides server
+
+    These tests rely on a Fides connector config that is configured to
+    connect to the main Fides server running in the
+    docker compose test environment.
+
+    This is not a realistic use case, but it can be used to verify
+    the core FidesConnector functionality, without relying on more than
+    one Fides server instance to be running.
+    """
+
+    def test_test_connection(self, test_fides_connector: FidesConnector):
+        assert test_fides_connector.test_connection() == ConnectionTestStatus.succeeded
+
+    def test_retrieve_data(
```

Comment:
i can dig into this test a bit more, but i'd like to discuss approaches for integration testing generally before going down that path